ducktape/tests/test.py

```diff
@@ -151,16 +150,19 @@ def __init__(self, **kwargs):
         self.injected_args = kwargs.get("injected_args", None)
         self.ignore = kwargs.get("ignore", False)
 
-        self.services = kwargs.get("services", ServiceRegistry())
+        self.services = ServiceRegistry()
 
         # dict for toggling service log collection on/off
-        self.log_collect = kwargs.get("log_collect", {})
+        self.log_collect = {}
 
     def __repr__(self):
         return "<module=%s, cls=%s, function=%s, injected_args=%s>" % \
                (self.module, self.cls_name, self.function_name, str(self.injected_args))
 
-    def copy(self, **kwargs):
+    def copy_constructor(self, **kwargs):
```

Comment:
Why did the name change? Not that big a deal, but (despite my writing), I am a fan of concise names and actually putting "constructor" in the names of constructors is unusual (and also, this is not a constructor). It also required changes a bunch of places that muddy the git history :)